_search-plugins/star-tree-index.md

```diff
@@ -0,0 +1,189 @@
+
+Star-tree index can be used to perform faster aggregations. Consider the following criteria and features when deciding to use a star-tree index:
+
+- Star-tree indexes natively support multi-field aggregations.
+- Star-tree indexes will be created in real-time as part of the indexing process, so the data in a star-tree will always be up to date.
+- A star-tree index consolidates data making the index efficient at paging and using less IO for search queries.
+
+## Limitations
+
+Star-tree indexes contain the following limitations:
+
+- Star-tree index should only be used on indexes whose data is not updated or deleted, as updates and deletions are not accounted for in a star-tree index.
+- A star-tree index can be used for aggregation queries only if the fields queried is a subset of the star-trees dimensions and the aggregated fields are a subset of a star-trees metrics.
+- After the star-tree index is enabled for an index, it cannot be disabled. In order to disable the star-tree index, the data in the index must be re-indexed without the star-tree mapping. Furthermore, changing a star-tree configuration will also require a re-index operation.
+- [Multi-values/array values]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/index/#arrays) are not supported
+- Only [limited queries and aggregations](#supported-query-and-aggregations) are supported. Support for more features will be released in upcoming versions.
+- The cardinality of the dimensions should not be very high (like `_id` fields). High cardinality leads to higher storage usage and higher query latency.
+
+## Star-tree index structure
+
+The following image illustrates a standard star-tree index structure.
+
+<img src="{{site.url}}{{site.baseurl}}/images/star-tree-index.png" alt="A star-tree index containing two dimensions and two metrics" width="700">
+
+Sorted and aggregated star-tree documents are backed by `doc-values` in an index. Values follow the following patternL
+
+- The values are sorted based on the order of `ordered_dimension`, in the previous example, first by `status` and then by `port` for each of the status.
+- For each unique dimension value combination, the aggregated values for all the metrics such as `avg(size)` and `count(requests)` are pre-computed during ingestion time.
+
+### Leaf nodes
+
+Each node in the star-tree index points to a range of star-tree documents. Nodes can be further split into child nodes based on the[max_leaf_docs configuration]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/star-tree/#star-tree-configuration-parameters). The number of documents a leaf node points to is less than or equal to the number `max_leaf_docs` set. This ensures the maximum number of documents that traverse across nodes to get to the aggregated value is at most the `max_leaf_docs`, which provides predictable latency.
+
+### Star nodes
+
+`star nodes (*)` fetch aggregated document whenever applicable during query time.
```

Comment:
```suggestion
`star nodes (*)` fetch aggregated documents during query time.
```

Reply:
Tried to come up with a better explanation based on the Pinot documentation, https://docs.pinot.apache.org/basics/indexing/star-tree-index#index-generation